src/creo2urdf/src/Creo2Urdf.cpp

```diff
@@ -277,34 +275,33 @@ void Creo2Urdf::OnCommand() {
             if (config["reverseRotationAxis"].IsDefined() && 
                 config["reverseRotationAxis"].Scalar().find(joint_name) != std::string::npos)
             {
-                axis = axis.reverse();
+                direction = direction.reverse();
             }
 
-            auto urdf_child_link_name = getRenameElementFromConfig(child_link_name);
-
+            auto urdf_parent_link_name = getRenameElementFromConfig(parent_link_name);
+            
+            auto axis_origin = iDynTree::Position::Zero();
 
-            // FIXME
-            // It does something but the transform is somehow wrong, in case of proper definition of the csys oldChild_H_newChild is the identity
-            // Once we found the proper transform we should updatea also the spatial inertia
+            iDynTree::Axis idyn_axis{ direction, parentLink_H_childLink.getPosition() };
 
-            parent_H_child = parent_H_child * oldChild_H_newChild;
-            auto link_H_collision_solid_shape = idyn_model.collisionSolidShapes().getLinkSolidShapes()[idyn_model.getLinkIndex(urdf_child_link_name)][0]->getLink_H_geometry();
-            auto link_H_visual_solid_shape    = idyn_model.visualSolidShapes().getLinkSolidShapes()[idyn_model.getLinkIndex(urdf_child_link_name)][0]->getLink_H_geometry();
-
-            idyn_model.collisionSolidShapes().getLinkSolidShapes()[idyn_model.getLinkIndex(urdf_child_link_name)][0]->setLink_H_geometry(oldChild_H_newChild.inverse() * link_H_collision_solid_shape);
-            idyn_model.visualSolidShapes().getLinkSolidShapes()[idyn_model.getLinkIndex(urdf_child_link_name)][0]->setLink_H_geometry(oldChild_H_newChild.inverse() *  link_H_visual_solid_shape);
+            // Check if the axis is aligned with the link frame
+            if ( parent_link_frame == "CSYS" || idyn_axis.getDistanceBetweenAxisAndPoint(iDynTree::Position::Zero()) > 1e-7 ) {
+                printToMessageWindow("The axis " + axis_name + " of "+ joint_name+ " is not aligned with the link frame " + parent_link_frame + ", the axis will be moved to the link frame in order to be urdf-compliant", c2uLogLevel::WARN);
```

Comment:
I am a bit dubious about printing this as a warning. In the case of iCub 2.7, this is expected, and adding warning that can't be fixed is typically a good way to ensure that people actually do not check warnings. Anyhow, this is non blocking.

Reply:
Actually that if is not correct, because that distance will be never > 0, I put the PR in draft until I fix it